src/main/java/uk/ac/ebi/atlas/experimentpage/markergenes/MarkerGenesDao.java

```diff
@@ -32,74 +41,61 @@ public MarkerGenesDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
                 namedParameters,
                 (resultSet, rowNumber) -> MarkerGene.create(
                         resultSet.getString("gene_id"),
-                        resultSet.getInt("k_where_marker"),
-                        resultSet.getInt("cluster_id_where_marker"),
+                        resultSet.getString("k_where_marker"),
+                        resultSet.getString("cluster_id_where_marker"),
                         resultSet.getDouble("marker_p_value"),
-                        resultSet.getInt("cluster_id"),
+                        resultSet.getString("cluster_id"),
                         resultSet.getDouble("median_expression"),
                         resultSet.getDouble("mean_expression")));
     }
 
     private static final String SELECT_DISTINCT_KS_WITH_MARKER_GENES =
-            "SELECT DISTINCT k_where_marker " +
-                    "FROM scxa_marker_gene_stats " +
-                    "WHERE experiment_accession = :experiment_accession AND marker_p_value < 0.05 " +
+            "SELECT DISTINCT h.variable as k_where_marker " +
```

Comment:
@upendrakumbham I have my doubts on whether this is the correct replacement for "k_where_marker".  I would expect m.cell_group_id to the be correct replacement for k_where_marker, and not h.variable.

Reply:
No, forget what I wrote, this is correct.